### DISCUSSION TITLE
What is the MAUI future ?

### DISCUSSION THREAD

**Discussion by Transis-Felipe** [dotnet/maui#18413] [General] [answered]
I'm worried about the future of MAUI.
Last weeks there's so little PRs.

@PureWeen @StephaneDelcroix @jonathanpeppers  aren't working with MAUI anymore ?

After the discontinuation of VS for MAC, MS said that is commited with VS and MAUI, but it's not what we see here.
Should I put resources on migration of Xamarin projects to MAUI or should I move to another plataform ?

**Comment by jonathanpeppers** [comment, marked as answer]
I think you are experiencing what it's like when something is about to ship: .NET 8. We are all testing the product.

**Reply by jonathanpeppers** [reply]
Just to add a second data-point, dotnet/maui was the second highest contributed .NET foundation repository in November:

* https://dnfprojects.org/!/Nov-01-2023/Nov-30-2023/microsoft/pr

(Note that the xamarin-android and xamarin-macios repositories are also under the "MAUI" umbrella)

**Reply by Xyncgas** [reply]
You have to look at the contributions to see how many of those are fixing issues

**Comment by MhAllan** [comment]
I predicted the death of many techs in the past: Asp Webforms, VB, IE, UWP, Angular, Javascript, CSS, Html, Windows Phone, Surface Duo. Clearly frontend stack survived despite how terrible it is.

And I can say, something like Maui has very good design that is inspired by WPF. But, it has terrible quality (they will beat Xamarin count of bugs soon), and terrible momentum (this is since 2014 and not working), and terrible prioritization (I won't be surprised that after .net 8 they will prioritize something no one care about, like android fast memory profiler 😅). and it's team is small and struggling to get simple things (like borders of a button) to work properly and that is not their fault but MS fault that it doesn't want to invest in this. 

Result is: This is going to die or it is dead just waiting courage to announce it.

**Reply by MicroControllerTeamProject** [reply]
you're forgetting about the prince of all failures!!!   ....SILVERLIGHT (echo) :-)))

**Reply by Xyncgas** [reply]
Xamarin was painful enough, navigating in a different style of url than the web, the 'big thinking' required out of me to write a software by myself to be caring about 'UI chores' was not what I was getting paid for and most .NET developer are low to mid income

Because Xamarin is really liking the idea of 'Binding' and 'XAML' markups and 'Shells', but I wanted a simple way to put something on screen and removing it by ```Control.Remove``` while being performant

**Comment by yyliveyy** [comment]
I am thinking about a question, what does Windows desktop development use? Using MAUI? But it cannot support desktop level application development. Using WPF or WINFORM? But Microsoft is discontinuing support for these technologies.

**Comment by maonaoda** [comment]
Due to the poor quality, it is foreseeable that there will be another alternative to maui in the future, just like maui replaces Xamarin. This is MS

**Reply by orange-pig** [reply]
MS: The New One! (...WindowPhone,UWP,MAUI)
Developer: ohhhhhhhhhh!
MS: Looks like you don't like it very much, throw it away! The Next One!
Developer: I know, He always is.

**Reply by MhAllan** [reply]
my guess is MS will give up. They wanted it to boost Azure, but it didn't do anything.

**Comment by LUJIAN2020** [comment]
The most suitable option is avalonia

**Comment by NCLnclNCL** [comment]
Avalonia

**Comment by tjiakwokyung28** [comment]
maui future is running on harmonyOs next

**Comment by Xyncgas** [comment]
Maybe Microsoft should've let Xamarin live in the wild instead of buying it, now all xamarin related technology doesn't seem to work for the trends our users' stack is going.

It's not useful for building webapp, it's native component instead of unified UI on electrons etc, not saying unified experience is better it's the trends though.

It seems difficult trying to do the things other environment can do on native platforms too, WF/WPF on windows, Swift on MacOs using xcode and Java for android, you simply get a better experience using what everyone is using on serious projects.

The concept of Xamarin is interesting, I can abstract controls so I write a label and it displays a label on all devices and I can use all devices' capabilities instead of restrained by the compatibility of different devices' standards that are yet unified, enjoying features of .NET compiling tools such as NativeAOT

It is not enough to use it for developing apps for Microsoft themselves apparently, though you can argue a reasoning behind the decision was to bet on competitor's technology

**Comment by tjiakwokyung28** [comment]
after 2 years i'am development using maui, maui is the best technology, but the only drawback is that it doesn't support HarmonyOS which in the future will have a market share of billions of users

**Reply by NCLnclNCL** [reply]
harmonyOs is android ??

**Reply by Xyncgas** [reply]
![image](https://github.com/dotnet/maui/assets/35312807/1d6b3b45-6767-4ded-ba62-652ed82e134f)

https://github.com/dotnet/maui/discussions/18413#discussioncomment-7850011
https://github.com/dotnet/maui/discussions/15536#discussioncomment-7778001
https://github.com/dotnet/maui/discussions/19112

**Reply by Stedy59** [reply]
Short for "SpywareOS"

**Reply by ziomek64** [reply]
> Short for "SpywareOS" 

Any os not being open source is spy ware in one way or another most likely XD

**Reply by tjiakwokyung28** [reply]
we can not negatif thinking, be positif if you want be to success
